lib/datadog/opentelemetry/sdk/span_processor.rb

```diff
@@ -90,19 +92,35 @@ def start_datadog_span(span)
 
         # Some special attributes can override Datadog Span fields
         def span_arguments(span, attributes)
-          if attributes.key?('analytics.event')
-            attributes[Tracing::Metadata::Ext::Analytics::TAG_SAMPLE_RATE] =
-              attributes['analytics.event'] == 'true' ? 1 : 0
+          if attributes.key?('analytics.event') && (analytics_event = attributes['analytics.event']).respond_to?(:casecmp)
+            attributes[Tracing::Metadata::Ext::Analytics::TAG_SAMPLE_RATE] = analytics_event.casecmp('true') == 0 ? 1 : 0
           end
           attributes[Tracing::Metadata::Ext::TAG_KIND] = span.kind || 'internal'
 
-          kwargs = { tags: attributes, start_time: ns_to_time(span.start_timestamp) }
+          kwargs = { start_time: ns_to_time(span.start_timestamp) }
+
+          name = if attributes.key?('operation.name')
+                   attributes['operation.name']
+                 elsif (rich_name = Datadog::OpenTelemetry::Trace::Span.enrich_name(span.kind, attributes))
+                   rich_name.downcase
+                 else
+                   span.kind
+                 end
 
-          name = attributes.key?('operation.name') ? attributes['operation.name'] : span.name
-          kwargs[:resource] = attributes['resource.name'] if attributes.key?('resource.name')
+          kwargs[:resource] = attributes.key?('resource.name') ? attributes['resource.name'] : span.name
           kwargs[:service] = attributes['service.name'] if attributes.key?('service.name')
           kwargs[:type] = attributes['span.type'] if attributes.key?('span.type')
 
+          attributes.reject! { |key, _| OpenTelemetry::Trace::Span::DATADOG_SPAN_ATTRIBUTE_OVERRIDES.include?(key) }
+
+          # DEV: There's no `flat_map!`; we have to split it into two operations
```

Comment:
For later: Would it be possible to complete this with a single operation with `reduce` or `each_with_object`?

Reply:
This is an interesting line because I built it to support arbitrarily nested arrays, but that's actually not required for OTel, so I might simplify it later. But I wasn't able to in time for this PR.